components/src/dynamo/profiler/utils/dgdr_validate.py

```
@@ -124,53 +124,44 @@ def _validate_sla(sla: SLASpec) -> None:
 
+    # Planner
+    if is_planner_enabled(dgdr):
+        planner_cfg = dgdr.features.planner
+        # throughput scaling requires in-depth profiling data
+        if planner_cfg.enable_throughput_scaling:
+            planner_sweep_mode = planner_cfg.pre_deployment_sweeping_mode
+            if planner_sweep_mode == PlannerPreDeploymentSweepMode.None_:
+                raise ValueError(
+                    "pre_deployment_sweeping_mode in PlannerConfig cannot be 'none' when enable_throughput_scaling is enabled. "
+                    "Throughput-based scaling requires pre-deployment sweeping to generate engine performance data."
+                )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)